src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/watasks/controllers/PostTaskCancelByIdControllerTest.java

```diff
@@ -282,5 +293,36 @@ public void user_should_cancel_task_when_grant_type_challenged_and_permissions_r
         common.cleanUpTask(taskId);
     }
 
+    @Test
+    public void user_should_cancel_task_when_granular_permission_satisfied() {
+
+        TestVariables taskVariables = common.setupWATaskAndRetrieveIds("requests/ccd/wa_case_data.json", "processApplication");
+
+        common.setupHearingPanelJudgeForSpecificAccess(assignerCredentials.getHeaders(), taskVariables.getCaseId(), WA_JURISDICTION, WA_CASE_TYPE);
+        initiateTask(taskVariables, Jurisdiction.WA);
+
+        common.setupCFTOrganisationalRoleAssignmentForWA(granularPermissionCaseworkerCredentials.getHeaders());
+        String taskId = taskVariables.getTaskId();
+        Response resultAssignee = restApiActions.post(
+            "task/{task-id}/assign",
+            taskId,
+            new AssignTaskRequest(getAssigneeId(granularPermissionCaseworkerCredentials.getHeaders())),
+            assignerCredentials.getHeaders()
+        );
+
+        resultAssignee.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
+
+        Response result = restApiActions.post(
+            ENDPOINT_BEING_TESTED,
+            taskId,
+            granularPermissionCaseworkerCredentials.getHeaders()
+        );
+        result.then().assertThat()
+            .statusCode(HttpStatus.NO_CONTENT.value());
```

Comment:
check task state to verify updated to CANCEL?

Reply:
cftTaskState tested with 'pendingTermination'